From 8a18ec62f99ea4682f20bea087af17558780c6d4 Mon Sep 17 00:00:00 2001
From: Arzhan Kinzhalin <arzhan.i.kinzhalin@intel.com>
Date: Mon, 25 Sep 2017 17:41:48 +0000
Subject: [PATCH] Use p11-trust instead of trust.

Clear Linux provides a stub instead of 'trust' to avoid using it for
trust store management. The native p11-kit 'trust' command is provided
as 'p11-trust' instead.

This change:

* Adds 'trust-stub'.
* Changes 'trust' invocation in p11-kit to 'p11-trust'.

NB: The build instals 'trust-stub' as 'trust'. And 'trust' as
'p11-trust' in the package.

Signed-off-by: Arzhan Kinzhalin <arzhan.i.kinzhalin@intel.com>
---
 p11-kit/p11-kit.c |  2 +-
 trust-stub        | 11 +++++++++++
 2 files changed, 12 insertions(+), 1 deletion(-)
 create mode 100644 trust-stub

diff --git a/p11-kit/p11-kit.c b/p11-kit/p11-kit.c
index 7a90046..fabb230 100644
--- a/p11-kit/p11-kit.c
+++ b/p11-kit/p11-kit.c
@@ -76,7 +76,7 @@ p11_kit_trust (int argc,
 	args = calloc (argc + 2, sizeof (char *));
 	return_val_if_fail (args != NULL, 1);
 
-	args[0] = BINDIR "/trust";
+	args[0] = BINDIR "/p11-trust";
 	memcpy (args + 1, argv, sizeof (char *) * argc);
 	args[argc + 1] = NULL;
 
diff --git a/trust-stub b/trust-stub
new file mode 100644
index 0000000..56402cd
--- /dev/null
+++ b/trust-stub
@@ -0,0 +1,11 @@
+#!/usr/bin/bash
+
+cat <<EOF
+Clear Linux provides 'clrtrust' command to manage trust system-wide. Use
+'clrtrust --help' to view the available commands. Use of p11-kit 'trust' for
+trust store management is not supported and should be avoided.
+
+p11-kit's 'trust' is still available as 'p11-trust'.
+EOF
+
+exit 1
-- 
2.14.2

